client/header/activity-panel/panels/orders.js

```diff
@@ -265,29 +274,37 @@ export default compose(
 			return { orders: [], isError: true, isRequesting: false, orderStatuses };
 		}
 
-		if ( hasActionableOrders ) {
+		// Query the core Orders endpoint for the most up-to-date statuses.
+		const allOrdersQuery = {
+			page: 1,
+			per_page: QUERY_DEFAULTS.pageSize,
+			status: orderStatuses,
+		};
+		const actionableOrders = Array.from( getItems( 'orders', allOrdersQuery ).values() );
+
+		if ( hasActionableOrders && actionableOrders.length ) {
```

Comment:
I'm unable to see orders here since `hasActionableOrders` is false.  I'm thinking `hasActionableOrders` might not be needed here.  Since we've already queried for `actionableOrders` we can just check its length.

On that same note, we may want to update the unread indicator to use the same test so those aren't out of sync.

Reply:
Hmm.. I'm not seeing `hasActionableOrders` as `false`. It ultimate comes from the unread indicator function: https://github.com/woocommerce/woocommerce-admin/blob/71e7e5cbf46ee3ddac31452463ec527b12cbcf04/client/header/activity-panel/index.js#L148, https://github.com/woocommerce/woocommerce-admin/blob/71e7e5cbf46ee3ddac31452463ec527b12cbcf04/client/header/activity-panel/index.js#L283